tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -256,7 +258,8 @@ class EditorConsumer {
 
   virtual bool handleSourceText(StringRef Text) = 0;
 
-  virtual bool handleSerializedSyntaxTree(StringRef Text) = 0;
+  virtual bool handleSyntaxTree(swift::syntax::SourceFileSyntax SyntaxTree,
+                                std::unordered_set<unsigned> ReusedNodeIds) = 0;
```

Comment:
Why do we need to copy the syntax tree over? can we pass parameter by reference?

Reply:
I guess we could. We're just copying the root node though (at least I think so), so it's not a huge performance impact.